test/units/module_utils/network/common/test_utils.py

```diff
@@ -18,134 +18,170 @@
+
+    string_data = '123'
+    assert string_data == sort_list(string_data)
+
+def test_dict_diff():
+    base = dict(obj2=dict(), b1=True, b2=False, b3=False,
+                one=1, two=2, three=3, obj1=dict(key1=1, key2=2),
+                l1=[1, 3], l2=[1, 2, 3], l4=[4],
+                nested=dict(n1=dict(n2=2)))
+
+    other = dict(b1=True, b2=False, b3=True, b4=True,
+                 one=1, three=4, four=4, obj1=dict(key1=2),
+                 l1=[2, 1], l2=[3, 2, 1], l3=[1],
+                 nested=dict(n1=dict(n2=2, n3=3)))
+
+    result = dict_diff(base, other)
+
+    # string assertions
+    assert 'one' not in result
+    assert 'two' not in result
+    assert result['three'] == 4
+    assert result['four'] == 4
+
+    # dict assertions
+    assert 'obj1' in result
+    assert 'key1' in result['obj1']
+    assert 'key2' not in result['obj1']
+
+    # list assertions
+    assert result['l1'] == [2, 1]
+    assert 'l2' not in result
+    assert result['l3'] == [1]
+    assert 'l4' not in result
+
+    # nested assertions
+    assert 'obj1' in result
+    assert result['obj1']['key1'] == 2
+    assert 'key2' not in result['obj1']
+
+    # bool assertions
+    assert 'b1' not in result
+    assert 'b2' not in result
+    assert result['b3']
+    assert result['b4']
+
+def test_dict_merge():
+    base = dict(obj2=dict(), b1=True, b2=False, b3=False,
+                one=1, two=2, three=3, obj1=dict(key1=1, key2=2),
+                l1=[1, 3], l2=[1, 2, 3], l4=[4],
+                nested=dict(n1=dict(n2=2)))
+
+    other = dict(b1=True, b2=False, b3=True, b4=True,
+                 one=1, three=4, four=4, obj1=dict(key1=2),
+                 l1=[2, 1], l2=[3, 2, 1], l3=[1],
+                 nested=dict(n1=dict(n2=2, n3=3)))
+
+    result = dict_merge(base, other)
+
+    # string assertions
+    assert 'one' in result
+    assert 'two' in result
+    assert result['three'] == 4
+    assert result['four'] == 4
+
+    # dict assertions
+    assert 'obj1' in result
+    assert 'key1' in result['obj1']
+    assert 'key2' in result['obj1']
+
+    # list assertions
+    assert result['l1'] == [1, 2, 3]
+    assert 'l2' in result
+    assert result['l3'] == [1]
+    assert 'l4' in result
+
+    # nested assertions
+    assert 'obj1' in result
+    assert result['obj1']['key1'] == 2
+    assert 'key2' in result['obj1']
+
+    # bool assertions
+    assert 'b1' in result
+    assert 'b2' in result
+    assert result['b3']
+    assert result['b4']
+
+def test_conditional():
+    assert conditional(10, 10)
+    assert conditional('10', '10')
+    assert conditional('foo', 'foo')
+    assert conditional(True, True)
+    assert conditional(False, False)
+    assert conditional(None, None)
+    assert conditional("ge(1)", 1)
+    assert conditional("gt(1)", 2)
+    assert conditional("le(2)", 2)
+    assert conditional("lt(3)", 2)
+    assert conditional("eq(1)", 1)
+    assert conditional("neq(0)", 1)
+    assert conditional("min(1)", 1)
```

Comment:
I think these should be `is True` since the `.assertTrue` being replaced would fail if on something like `"foo"` which is truthy. So the `assert "foo"` would pass where `self.assertTrue("foo")` wouldn't.

Reply:
Oh. This is a lie, ignore that comment. 

> Note that this is equivalent to `bool(expr) is True` and not to `expr is True`

https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertTrue